### PR TITLE
test/cluster: write up instructions for manual clustering demo

### PR DESCRIPTION
**Pull request by benesch**
To share with prospects who are interested in clustering.

### Motivation

  * This PR adds a known-desirable feature.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR adds a release note for any [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/user/content/release-notes.md#what-changes-require-a-release-note).

### CODE REVIEW

File: test/cluster/README.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+Clustering is an **enterprise feature.** Per the terms of our
+[LICENSE](/LICENSE), you may not run Materialize in clustered mode without
+an enterprise license.
+
+⚠️ **WARNING!** ⚠️ Materialize clustering is experimental! Do not use this in
+production.
+
+## Getting started
+
+To run this demo, you'll need three separate machines:
```
Comment (antiguru) [inline test/cluster/README.md:17]: Nit: Docker runs this happily on a single machine. Virtual machines?

File: test/cluster/README.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+You can run dataflow clusters of arbitrary size. Suppose you want to run a
+cluster with *N* dataflow nodes and *W* worker threads per node. To launch
+the *I*th dataflow node, run:
+
+```
+docker run -p 2101:2101 -p 6876:6876 materialize/dataflowd:latest \
+    --workers <W> \
+    --processes <N> --process <I> \
+    --hosts dataflow1:2101 dataflow2:2101 ... dataflow<N>:2101
```
Comment (antiguru) [inline test/cluster/README.md:56]: Does docker name the images automatically like `dataflow1` and so on?

**Review by antiguru — APPROVED** [review]
LGTM, thanks!

**Comment by benesch** [comment]
Ah, to answer your questions: no, Docker doesn't help at all with naming the hosts. The use of Docker is simply because it's the easiest way to get access to those binaries. I added a paragraph to explain that—WDYT?

**Comment by antiguru** [comment]
Looks good, thanks!

**Pull request merged**